cocos/audio/android/AudioDecoder.cpp

```diff
@@ -264,16 +270,19 @@ bool AudioDecoder::decodeToPcm()
     decDest.pLocator = (void *) &decBuffQueue;
     decDest.pFormat = (void *) &pcm;
 
-    /* Create the audio player */
-    result = (*_engineItf)->CreateAudioPlayer(_engineItf, &player, &decSource, &decDest,
-                                              NUM_EXPLICIT_INTERFACES_FOR_PLAYER, iidArray,
-                                              required);
-    SL_RETURN_VAL_IF_FAILED(result, false, "CreateAudioPlayer failed");
-
-    _playObj = player;
-    /* Realize the player in synchronous mode. */
-    result = (*player)->Realize(player, SL_BOOLEAN_FALSE);
-    SL_RETURN_VAL_IF_FAILED(result, false, "Realize failed");
+    {
+        std::lock_guard<std::mutex> lk(__SLPlayerMutex);
+        /* Create the audio player */
+        result = (*_engineItf)->CreateAudioPlayer(_engineItf, &player, &decSource, &decDest,
+                                                  NUM_EXPLICIT_INTERFACES_FOR_PLAYER, iidArray,
+                                                  required);
+        SL_RETURN_VAL_IF_FAILED(result, false, "CreateAudioPlayer failed");
+
+        _playObj = player;
+        /* Realize the player in synchronous mode. */
+        result = (*player)->Realize(player, SL_BOOLEAN_FALSE);
+        SL_RETURN_VAL_IF_FAILED(result, false, "Realize failed");
+    }
```

Comment:
If without `}` after `->Realize`, multi-thread decoding will be broken. Although, it does use multi-threads, all tasks in thread pool will wait previous decoding task to finish, then start. So removing this brace will make no sense.